File: Programmers/JavaScript/collatz.js
var num = 16
var answer = 0


while (true) {
    if (num == 1) {
        break
    }
    else if (answer == 500) {
        answer = -1
        break
    }

    if (num % 2 == 0) {
        num /= 2
    }
    else {
        num = num * 3 + 1
    }
    answer++
}

console.log(answer)

File: Programmers/Python/collatz.py
n = 16
answer = 0

while True:
    if answer >= 500:
        answer = -1
        break

    if n == 1:
        break

    if n % 2 == 0:
        n /= 2
    else:
        n = n * 3 + 1

    answer += 1

print(answer)
